src/routes/formsgSiteCreation.ts

```diff
@@ -157,9 +166,83 @@ export class FormsgRouter {
     await mailer.sendMail(email, subject, html)
   }
 
+  cloneSiteToEfs: RequestHandler<
+    never,
+    Record<string, never>,
+    { data: { submissionId: string } },
+    never,
+    { submission: DecryptedContent }
+  > = async (req, res) => {
+    // 1. Extract arguments
+    const { submissionId } = req.body.data
+    const { responses } = res.locals.submission
+    // NOTE: This is validated by formsg to be of domain `@open.gov.sg`;
+    // hence, not revalidating here
+    const requesterEmail = getField(responses, "Email") as string
+    // NOTE: The field is required by our form so this cannot be empty or undefined
+    const githubRepoName = getField(responses, "Github Repo Name") as string
+
+    logger.info(
+      `${requesterEmail} requested for ${githubRepoName} to be cloned onto EFS`
+    )
+
+    this.gitFileSystemService
+      .clone(githubRepoName)
+      .map((path) => {
+        logger.info(`Cloned ${githubRepoName} to ${path}`)
+        this.sendCloneSuccess(
+          requesterEmail,
+          githubRepoName,
+          submissionId,
+          path
+        )
+      })
+      .mapErr((err) => {
+        logger.error(
+          `Cloning repo: ${githubRepoName} to EFS failed with error: ${JSON.stringify(
+            err
+          )}`
+        )
+        this.sendCloneError(
+          requesterEmail,
+          githubRepoName,
+          submissionId,
+          err.message
+        )
+      })
+  }
+
+  sendCloneSuccess = async (
+    requesterEmail: string,
+    githubRepoName: string,
+    submissionId: string,
+    path: string
+  ) => {
+    const subject = `[Isomer] Clone site ${githubRepoName} SUCCESS`
+    const html = `<p>Isomer site ${githubRepoName} was cloned successfully to EFS path: ${path}. (Form submission id [${submissionId}])</p>`
+    await mailer.sendMail(requesterEmail, subject, html)
+  }
+
+  async sendCloneError(
+    requesterEmail: string,
+    githubRepoName: string,
+    submissionId: string,
+    message: string
+  ) {
+    const subject = `[Isomer] Clone site ${githubRepoName} FAILURE`
+    const html = `<p>Isomer site ${githubRepoName} was <b>not</b> cloned successfully. Cloning failed with error: ${message} (Form submission id [${submissionId}])</p>`
+    await mailer.sendMail(requesterEmail, subject, html)
+  }
+
   getRouter() {
     const router = express.Router({ mergeParams: true })
 
+    router.post(
+      "/clone-site",
+      attachFormSGHandler("XO8YmgUbENTEftCEvnZRxM1Nb+wpERTfEj/w+4kPink="),
```

Comment:
also to confirm this middleware verifies the incoming request is actually from form sg right?